### ISSUE TITLE
environment.yml

### ISSUE THREAD

**Issue by Ystartff** [zqgao22/HIGH-PPI#9]
Hi! I ran your conda env create -f environment.yml and found that the version incompatibility does not cause the creation of a virtual environment, can you provide some solutions?

**Comment by horacehht** [comment]
Me too!!! conda says it found conflict

**Comment by horacehht** [comment]
@Ystartff 
I encounter this situation too. Later, I choose to **manully install packages** mentioned in the yaml file like `conda install dgl=version` or `pip install tensorboardX==version`.
And I found that you don't need to install the packages  one by one, which looks ridiculous.
You just need to install **pytorch, cudatoolikit and tensorboardX**. Then you install the **dgl** with appropriate version corresponding to the pytorch. What's more, follow the instructions in readme.md to install **torch-scatter, torch-spline-conv, torch-sparse and torch -cluster** according to the version of torch and cuda.
I download these packages on ubuntu using these commands.(Just an example, my torch version is 1.13 and cuda version is 11.7)
wget https://data.pyg.org/whl/torch-1.13.0%2Bcu117/torch_spline_conv-1.2.1%2Bpt113cu117-cp39-cp39-linux_x86_64.whl
wget https://data.pyg.org/whl/torch-1.13.0%2Bcu117/torch_sparse-0.6.15%2Bpt113cu117-cp39-cp39-linux_x86_64.whl
wget https://data.pyg.org/whl/torch-1.13.0%2Bcu117/torch_scatter-2.1.0%2Bpt113cu117-cp39-cp39-linux_x86_64.whl
wget https://data.pyg.org/whl/torch-1.13.0%2Bcu117/torch_cluster-1.6.0%2Bpt113cu117-cp39-cp39-linux_x86_64.whl

Finnally, you pip install these four packages and torch-geometric.
When all is done, you can run the project. The version change doesn't affect the running of the project

**Comment by Ystartff** [comment]
@horacehht I also completed this problem through manual installation. In the process of copying the code, the fluctuations were great every time I experimented. Have you solved this problem?

**Comment by horacehht** [comment]
@Ystartff Sorry to tell you that I just experimented once. I will conduct multiple experiments over the next few days to observe, and if the same conclusions are drawn, I will inform you as soon as possible.

**Comment by Ystartff** [comment]
Thank you for your help, thank you very much

**Comment by horacehht** [comment]
> @horacehht I also completed this problem through manual installation. In the process of copying the code, the fluctuations were great every time I experimented. Have you solved this problem?

@Ystartff Yes, after I set 9 seeds to run HIGH-PPI, I found that the fluctuations were great. My seeds were [7, 17, 27, 37, 47, 57, 67, 87, 3407]

**Comment by horacehht** [comment]
> @horacehht I also completed this problem through manual installation. In the process of copying the code, the fluctuations were great every time I experimented. Have you solved this problem?

@Ystartff Hello. I think that I found the reasons. In model_train.py, the default 'factor' argument in lr_scheduler and the batch size is set to 0.5 and 11000, respectively. The default batch size exceeds the dataset length, which implies that the entire dataset is used during training initiation. There is a convention that larger batch size needs larger large learning rate. I observed the learning rate during training and found that  learning rate drops sharply, which is not beneficial for the tasks.
So, you can change the default 'factor'. Accroding to my observation, factor set as 0.8 wiil produce a good performance.

**Comment by chenxofhit** [comment]
It appears that there may be issues with the yml file, as I encountered significant difficulties during the installation process. It seems that manual installation of individual packages may be the most viable approach in this situation.

| WARNING conda.models.version:get_matcher(540): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.6.0.*, but conda is ignoring the .* and treating it as 1.6.0
WARNING conda.models.version:get_matcher(540): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.8.0.*, but conda is ignoring the .* and treating it as 1.8.0
WARNING conda.models.version:get_matcher(540): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.9.0.*, but conda is ignoring the .* and treating it as 1.9.0
WARNING conda.models.version:get_matcher(540): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.7.1.*, but conda is ignoring the .* and treating it as 1.7.1

Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
